lib/active_model/serializer/type.rb

```diff
@@ -17,7 +17,7 @@ module ClassMethods
         #   class AdminAuthorSerializer < ActiveModel::Serializer
         #     type 'authors'
         def type(type)
-          self._type = type
+          self._type = type ? type.to_s : nil
```

Comment:
Because of [this](https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model/serializer/adapter/fragment_cache.rb#L98), we might call `serializer.type(nil)` and then `nil.to_s` results in `""`. I tried `self._type = type if type` at first but then two specs were failling out of nowhere. It looked like something related to the cache but I could not really figure it out.